src/components/Main/Main.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { useContext, useState } from 'react';
+
+import './Main.scss';
+import { TodoList } from '../TodoList';
+import { DispatchContext, FilterContext, StateContext } from '../TodosContext';
+import { Status } from '../../types/Status';
+import { Todo } from '../../types/Todo';
+
+export const Main: React.FC = () => {
+  const todos = useContext(StateContext);
+  const dispatch = useContext(DispatchContext);
+  const filterStatus = useContext(FilterContext);
+
+  const [isChecked, setIsChecked] = useState(false);
+
+  let filteredTodos: Todo[] = [];
+
+  switch (filterStatus) {
+    case Status.All:
+      filteredTodos = [...todos];
+      break;
+
+    case Status.Active:
+      filteredTodos = [...todos].filter(todo => !todo.completed);
```

Comment:
```suggestion
      filteredTodos = todos.filter(todo => !todo.completed);
```
filter doesn't change the array, so this copy is redundant